### PR TITLE
Fixing ECDS documentation

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
Fixing ECDS documentation

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/service/extension/v3/config_discovery.proto [hunk 1]
```diff
@@ -18,6 +18,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
...
+// :ref:`Extension Config Discovery Service (ECDS) API <config_overview_extension_discovery>`.
+// The overall extension config discovery service works like this:
+//
+// 1. Envoy receives the ECDS configuration either from LDS or from static config, which has any
+//    number of listener filters and HTTP filters configured as ECDS filter. The actual
+//    configuration of these filters needs to be fetched from the xDS server.
+// 2. Envoy then sends the DiscoveryRequest message to the xDS server to retrieve the configuration.
+// 3. xDS server sends back the DiscoveryResponse to Envoy with the actual extension configuration.
+// 4. Envoy instantiates the filter factories with the received configuration.
+// 5. Next time when Envoy processing traffic, it builds the filter chains with the new configuration.
```
Comment (adisuissa) [inline api/envoy/service/extension/v3/config_discovery.proto:32]:
```suggestion
// 
// Note: Filters that are configured using ECDS are warmed. For more details see <ref ExtensionConfigSource api>.
```

**Review by adisuissa — COMMENTED** [review]
Thanks for working on this. Left a few suggestions.
cc @htuch for more suggestions.

File: api/envoy/service/extension/v3/config_discovery.proto [hunk 2]
```diff
@@ -18,6 +18,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: Extension config discovery service (ECDS)]
 
+// A service supports dynamically discover extension configuration and applies to
+// Envoy. Currently, ECDS is supported in HTTP filter and Listener filter. Please check
```
[the remark this answers is not in the document]
Comment (yanjunxiang-google) [reply api/envoy/service/extension/v3/config_discovery.proto]: Done! Thanks a lot for the detail comments. Please take a look the latest change.

File: api/envoy/service/extension/v3/config_discovery.proto [hunk 3]
```diff
@@ -18,6 +18,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
...
+// :ref:`Extension Config Discovery Service (ECDS) API <config_overview_extension_discovery>`.
+// The overall extension config discovery service works like this:
```
[the remark this answers is not in the document]
Comment (yanjunxiang-google) [reply api/envoy/service/extension/v3/config_discovery.proto]: done

File: api/envoy/service/extension/v3/config_discovery.proto [hunk 4]
```diff
@@ -18,6 +18,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
...
+//
+// 1. Envoy receives the ECDS configuration either from LDS or from static config, which has any
+//    number of listener filters and HTTP filters configured as ECDS filter. The actual
+//    configuration of these filters needs to be fetched from the xDS server.
```
[the remark this answers is not in the document]
Comment (yanjunxiang-google) [reply api/envoy/service/extension/v3/config_discovery.proto]: Done

File: api/envoy/service/extension/v3/config_discovery.proto [hunk 5]
```diff
@@ -18,6 +18,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
...
+// 2. Envoy then sends the DiscoveryRequest message to the xDS server to retrieve the configuration.
+// 3. xDS server sends back the DiscoveryResponse to Envoy with the actual extension configuration.
```
[the remark this answers is not in the document]
Comment (yanjunxiang-google) [reply api/envoy/service/extension/v3/config_discovery.proto]: done

File: api/envoy/service/extension/v3/config_discovery.proto [hunk 6]
```diff
@@ -18,6 +18,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
...
 
+// A service supports dynamically discover extension configuration and applies to
+// Envoy. Currently, ECDS is supported in HTTP filter and Listener filter. Please check
+// :ref:`Extension Config Discovery Service (ECDS) API <config_overview_extension_discovery>`.
+// The overall extension config discovery service works like this:
+//
+// 1. Envoy receives the ECDS configuration either from LDS or from static config, which has any
+//    number of listener filters and HTTP filters configured as ECDS filter. The actual
+//    configuration of these filters needs to be fetched from the xDS server.
+// 2. Envoy then sends the DiscoveryRequest message to the xDS server to retrieve the configuration.
```
[the remark this answers is not in the document]
Comment (yanjunxiang-google) [reply api/envoy/service/extension/v3/config_discovery.proto]: done

File: api/envoy/service/extension/v3/config_discovery.proto [hunk 7]
```diff
@@ -18,6 +18,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
...
+// 3. xDS server sends back the DiscoveryResponse to Envoy with the actual extension configuration.
+// 4. Envoy instantiates the filter factories with the received configuration.
```
[the remark this answers is not in the document]
Comment (yanjunxiang-google) [reply api/envoy/service/extension/v3/config_discovery.proto]: done

File: api/envoy/service/extension/v3/config_discovery.proto (hunk 1 above)
Comment (yanjunxiang-google) [reply api/envoy/service/extension/v3/config_discovery.proto]: done

**Review by htuch — APPROVED** [review]

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**